src/session/mod.rs

```diff
@@ -0,0 +1,248 @@
+// This file is part of Astarte.
+//
+// Copyright 2025 SECO Mind Srl
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+// SPDX-License-Identifier: Apache-2.0
+
+//! Handles the storage of the current introspection to maintain
+//! a persistent session with the Astarte MQTT server.
+
+use std::future::Future;
+
+use itertools::Itertools;
+
+use crate::{error::DynError, interfaces::Interfaces, Interface};
+
+mod sqlite;
+
+/// Interface data associated with the astarte introspection.
+#[derive(Debug, Clone, PartialOrd, Ord, Hash)]
```

Comment:
```suggestion
#[derive(Debug, Clone, Copy, PartialEq, Eq, PartialOrd, Ord, Hash)]
```

Reply:
I don't know if `PartialEq` is possible or would conflict with the manual impl.

But we should derive copy